### ISSUE TITLE
'file' and 'file_list' field types do not work for the Taxonomy page - edit-tags.php

### ISSUE THREAD

**Issue by ADC07** [WebDevStudios/CMB2#823]
The following behavior and issues are applicable to both the 'file' and 'file_list' field types when using them on a taxonomy. 

### Expected Behavior:
When an user clicks on the browse button (i.e. 'Add or Upload File') the media-frame modal should appear, allowing the user to either choose one of the listed attachments or to upload and create a new attachment.

### Actual Behavior:
An error is triggered in the browser console:
```
media-views.min.js?ver=4.7:3 Uncaught TypeError: Cannot read property 'limitExceeded' of undefined
    at n.initialize (media-views.min.js?ver=4.7:3)
    at n.initialize (media-views.min.js?ver=4.7:3)
    at n.e.View (backbone.min.js?ver=1.2.3:1)
    at n.constructor (wp-backbone.min.js?ver=4.7:1)
    at n.constructor (media-views.min.js?ver=4.7:4)
    at n [as constructor] (backbone.min.js?ver=1.2.3:1)
    at n [as constructor] (backbone.min.js?ver=1.2.3:1)
    at new n (backbone.min.js?ver=1.2.3:1)
    at Object.wp.media (media-models.min.js?ver=4.7:1)
    at Object.cmb._handleMedia (cmb2.min.js?ver=2.2.3.1:1)
```

This is actually preceded by an error that is triggered upon page load:
```
media-audiovideo.min.js?ver=4.7:1 Uncaught TypeError: Cannot read property 'id' of undefined
    at Object.d.1../controllers/audio-details.js (media-audiovideo.min.js?ver=4.7:1)
    at e (media-audiovideo.min.js?ver=4.7:1)
    at a (media-audiovideo.min.js?ver=4.7:1)
    at media-audiovideo.min.js?ver=4.7:1
``` 

### FWIW:
Upon inspection I have discovered that both errors were triggered due to incomplete script-localisation data for the **plupload** script, more specifically I'm referring to the **_wpPluploadSettings** variable. This eluded me for quite some time as apparently the script itself was being enqueued and localised properly however this was far from being true as there was another method call **wp_enqueue_media** that further enhanced the localisation-data.

In short: perhaps the **wp_enqueue_media** call should be done earlier.

### Steps to reproduce (I have confirmed I can reproduce this issue on the trunk branch):

1.  Nothing special, just use either of the fields in the Taxonomy page

PS: I've tested this on both WP 4.5 as well as 4.7.

**Comment by jtsternberg** [comment]
Hmm works as expected for me and have used it for several different projects. You may need to test after deactivating all other plugins to see if it's a plugin conflict.

**Comment by ADC07** [comment]
Sorry, I think I forgot to mention one key aspect, namely that I was using CMB2 as a WP plugin and not directly including it in mine. Perhaps this is what differs?

**Comment by jtsternberg** [comment]
No, that wouldn't matter at all.

**Comment by ADC07** [comment]
Hmm... then it must be something else. I've just re-checked and apparently I do not have any enabled plugins in my WP installation.

**Comment by bradp** [comment]
Closing due to no movement here. Feel free to re-open if needed.
